pkg/tls/self_signed_cert.go

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) F5, Inc.
+//
+// This source code is licensed under the Apache License, Version 2.0 license found in the
+// LICENSE file in the root directory of this source tree.
+
+// Package gencert generates self-signed TLS certificates.
+package selfsignedcert
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"log/slog"
+	"math/big"
+	"os"
+	"time"
+)
+
+const (
+	organization    = "F5 Inc."
+	filePermissions = 0o600
+)
+
+// nolint: revive
+func GenerateSelfSignedCert(certPath, keyPath string) (*tls.Certificate, error) {
+	// Return existing cert if cert exists
+	if _, certErr := os.Stat(certPath); certErr == nil {
+		if _, keyErr := os.Stat(keyPath); keyErr == nil {
+			cert, err := ReturnExistingCert(certPath, keyPath)
+			if err != nil {
+				return nil, err
+			}
+
+			return cert, nil
+		}
+	}
+
+	priv, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if err != nil {
+		return nil, fmt.Errorf("failed to generate private key: %w", err)
+	}
+
+	template := x509.Certificate{
+		SerialNumber: big.NewInt(1),
+		Subject: pkix.Name{
+			Organization: []string{organization},
+		},
+		NotBefore:             time.Now(),
+		NotAfter:              time.Now().Add(365 * 24 * time.Hour),
+		KeyUsage:              x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature,
+		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
+		BasicConstraintsValid: true,
+	}
+
+	certDER, err := x509.CreateCertificate(rand.Reader, &template, &template, &priv.PublicKey, priv)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create certificate: %w", err)
+	}
+
+	// Marshal private key
+	keyBytes, err := x509.MarshalECPrivateKey(priv)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal private key: %w", err)
+	}
+
+	// Encode certificate and key to PEM
+	certPEM := pem.EncodeToMemory(&pem.Block{Type: "CERTIFICATE", Bytes: certDER})
+	keyPEM := pem.EncodeToMemory(&pem.Block{Type: "EC PRIVATE KEY", Bytes: keyBytes})
+
+	// Write the certificate to a file
+	err = os.WriteFile(certPath, certPEM, filePermissions)
+	if err != nil {
+		return nil, errors.New("failed to write certificate file")
+	}
+
+	// Write the private key to a file
+	err = os.WriteFile(keyPath, keyPEM, filePermissions)
+	if err != nil {
+		return nil, errors.New("failed to write key file")
+	}
+
+	// Create TLS certificate
+	cert, err := tls.X509KeyPair(certPEM, keyPEM)
+	if err != nil {
+		return nil, errors.New("failed to create TLS key pair")
+	}
+
+	return &cert, nil
+}
+
+func ReturnExistingCert(certPath, keyPath string) (*tls.Certificate, error) {
```

Comment:
```suggestion
func CreateTlsCert(certPath, keyPath string) (*tls.Certificate, error) {
```
Think this might be a better name for the function